depthai_ros_driver/include/depthai_ros_driver/depthai_base.hpp

```diff
@@ -42,39 +104,27 @@ class DepthAIBase : public Node {
     ~DepthAIBase() = default;
 
 private:
-    enum Stream : std::size_t {
-        // all images at the beginning
-        LEFT,
-        RIGHT,
-        RECTIFIED_LEFT,
-        RECTIFIED_RIGHT,
-        DISPARITY,
-        DISPARITY_COLOR,
-        DEPTH,
-        PREVIEW_OUT,
-        // compressed image streams
-        JPEG_OUT,
-        VIDEO,
-        // non-image streams
-        META_D2H,
-        META_OUT,
-        OBJECT_TRACKER,
-        // utility enums
-        END,
-        IMAGE_END = META_D2H,
-        UNCOMPRESSED_IMG_END = JPEG_OUT
-    };
-    std::array<std::string, Stream::END> _stream_name{"left", "right", "rectified_left", "rectified_right", "disparity",
-            "disparity_color", "depth", "previewout", "jpegout", "video", "meta_d2h", "metaout", "object_tracker"};
-    std::array<std::string, Stream::END> _topic_name{"left", "right", "rectified_left", "rectified_right", "disparity",
-            "disparity_color", "depth", "previewout", "jpeg", "mjpeg", "meta_d2h", "object_info", "object_tracker"};
 
+    std::set<std::string> _enabled_streams;
     std::array<std::unique_ptr<ros::Publisher>, Stream::END> _stream_publishers;
     std::array<std::unique_ptr<ros::Publisher>, Stream::IMAGE_END> _camera_info_publishers;
 
     std::array<std::unique_ptr<camera_info_manager::CameraInfoManager>, Stream::IMAGE_END> _camera_info_manager;
+
     std::unique_ptr<camera_info_manager::CameraInfoManager> _defaultManager;
-    ros::ServiceServer _camera_info_default;
+    // ros::ServiceServer _camera_info_default;
+
+    std::unique_ptr<pluginlib::ClassLoader<rr::Pipeline> > _pipeline_loader;
+
+    boost::shared_ptr<rr::Pipeline> _pipeline_plugin;
```

Comment:
Should we use a unique_ptr instead?

Reply:
return type of `_pipeline_loader->createInstance` is `template<class T>
boost::shared_ptr<T> ClassLoader<T>::createInstance(const std::string & lookup_name)`, 
and and I think this is shared with `_pipeline_loader`.

Reply:
There's a `createUniqueInstance` as well